docs/src/Guides/20 Extensions.md

```diff
@@ -1,248 +1,581 @@
-inside this extension, you can put whatever you would like. And upon loading, the contents are added to the bot.
+### Setup
+
+Extensions are effectively just another Python file that contains a class that inherits from an object called `Extension`,
+inside this extension.
+
+For example, this is a valid extension file:
 
 ```python
 from interactions import Extension
 
+class MyExtension(Extension):
+    pass
+```
 
-class SomeClass(Extension):
-    ...
+??? note "Differences from Other Python Discord Libraries"
+    If you come from another Python Discord library, you may have noticed that there's a lack of an `__init__` and `setup` function in this example.
+    They still do exist as functions you *can* use (as discussed later), but interactions.py will do the appropriate logic to handle extensions
+    without either of the two.
 
+    For example, the following does the exact same thing as the above extension file:
 
-def setup(bot):
-    # This is called by interactions.py so it knows how to load the Extension
-    SomeClass(bot)
+    ```python
+    from interactions import Extension
+
+    class MyExtension(Extension):
+        def __init__(self, bot):
+            self.bot = bot
+
+    def setup(bot):
+        # yes, the bot does not need to do any special logic - you just need to pass it into the extension
+        MyExtension(bot)
+    ```
+
+### Events and Commands
+
+You probably want extensions to do a little bit more than just exist though. Most likely, you want some events and commands
+in here. Thankfully, they're relatively simple to do. Expanding on the example a bit, a slash command looks like this:
+
+```python
+from interactions import Extension, slash_command, SlashContext
+
+class MyExtension(Extension):
+    @slash_command()
+    async def test(self, ctx: SlashContext):
+        await ctx.send("Hello world!")
 ```
-As you can see, there's one extra bit, a function called `setup`, this function acts as an entry point for interactions.py,
-so it knows how to load the extension properly.
 
-To load a extension, you simply add the following to your `main` script, just above `bot.start`:
+As you can see, they're almost identical to how you declare slash commands in your main bot file, even using the same decorator.
+The only difference is the `self` variable - this is the instance of the extension that the command is being called in, and is
+standard for functions inside of classes. Events follow a similar principal.
+
+interactions.py will automatically add all commands and events to the bot when you load the extension (discussed later),
+so you don't need to worry about that.
+
+#### Accessing the Bot
+
+You may have noticed that the `self` variable is being used in the example above. This is because the bot is passed to the extension
+when it's loaded and is stored in the `self.bot` variable. This means you can access the bot from anywhere in your extension,
```

Comment:
```suggestion
When an extension is loaded, the library automatically sets the `bot` for you. With this in mind, to access your client, use `self.bot`.

Please do note, if your convention is to use `self.client` this is fine and supported natively as well. 
```

Removed the blatant spoonfeed and added useful info... lets be real, we shouldnt explain basic class usage, our policy is we expect basic python knowledge